src/main.rs

```diff
@@ -392,9 +392,9 @@ fn main() -> Result<()> {
     perf!("run test_bins", start_time, use_color);
 
     start_time = std::time::Instant::now();
-    let input = if let Some(redirect_stdin) = &parsed_nu_cli_args.redirect_stdin {
+    let input = if !std::io::stdin().is_terminal() {
         trace!("redirecting stdin");
-        PipelineData::ByteStream(ByteStream::stdin(redirect_stdin.span)?, None)
+        PipelineData::ByteStream(ByteStream::stdin(Span::unknown())?, None)
```

Comment:
@cptpiepmatz Is it reasonable to use `Span::unknown` here? I actually want it to point to the executable in this case (so I guess it's not unknown, maybe we need to decouple "unknown" from "point at the executable on purpose"?), but downstream consumers _might_ end up passing this into an `IoError::new` at some point?

Reply:
I disliked the multiple meanings of `Span::unknown` for a long time but I don't know anything better yet. I could think of an abstraction for it but that's out of scope here.